### PR TITLE
connlib: make dns request in a new task without blocking peers

### PR DESCRIPTION
**Pull request by conectado**
This required making `allow_access` `async` which is ugly, but we can fix it later like we did it with `set_peer_connection_request`, but doing this ASAP otherwise this would block the `peers_by_ip` struct and also block the executor a bunch of times and slow everything down.

### CODE REVIEW

File: rust/connlib/tunnel/src/control_protocol/gateway.rs [hunk 1]
```diff
@@ -101,9 +101,8 @@ where
                     return Err(Error::InvalidResource);
                 }
 
-                // TODO: we should make this async, this is acceptable for now though
```
Comment (jamilbk) [inline rust/connlib/tunnel/src/control_protocol/gateway.rs:104]: 🤣

File: rust/gateway/src/eventloop.rs [hunk 2]
```diff
@@ -169,20 +169,24 @@ impl Eventloop {
...
+                            if let Err(e) = sender
                                 .send(EgressMessages::ConnectionReady(ConnectionReady {
                                     reference,
                                     gateway_payload: GatewayResponse::ResourceAccepted(res),
                                 }))
                                 .await
-                        });
-                    }
+                            {
+                                tracing::warn!("Error while sending gateway response: {e:#?}");
```
Comment (jamilbk) [inline rust/gateway/src/eventloop.rs:186]: Are we sure this should be a warning? I.e. not expected to happen in normal operation.

**Review by jamilbk — APPROVED** [review]
Just a quick note about log level

**Review by thomaseizinger — APPROVED** [review]
Okay!

We will fix it soon :)

File: rust/gateway/src/eventloop.rs (hunk 2 above)
Comment (conectado) [reply rust/gateway/src/eventloop.rs:186]: it might be reconnecting

Comment (conectado) [reply rust/gateway/src/eventloop.rs:186]: we should buffer these messages so that they are not lost if gateway reconnects though, probably.

**Comment by jamilbk** [comment]
Will merge `main` into this when #3365 gets merged so that status checks are updated

**Pull request merged**